core/tauri/src/api/file/extract.rs

```diff
@@ -100,13 +101,8 @@ impl<'a, R: Read> Entry<'a, R> {
             }
           }
         } else {
-          let mut out_file = fs::File::create(into_path)?;
-          io::copy(&mut entry, &mut out_file)?;
-
-          // make sure we set permissions
-          if let Ok(mode) = entry.header().mode() {
-            set_perms(into_path, Some(&mut out_file), mode, true)?;
-          }
+          // handle files, symlinks, hard links, etc. and set permissions
+          entry.unpack(into_path)?;
```

Comment:
Using [`Entry::unpack_in`](https://docs.rs/tar/latest/tar/struct.Entry.html#method.unpack_in) will help prevent unpacking a file outside the destination.
```suggestion
          entry.unpack_in(into_path)?;
```

Reply:
We actually need to use unpack() here. The implementation is completely messed up, but it basically needs to extract `YourApp-0.1.1.app/Contents/MacOS/file` into `/tmp/tauri-update-blabla/ContentsMacOS/file` and later we rename `/tmp/tauri-update-blabla` to `/path/to/YourApp.app`

Reply:
oof

Reply:
But what we can do it check if the path has any parent component and error out... I don't think it should ever happen.